Configuration/RequestMiddlewares.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+return [
+    'frontend' => [
+        'bmack/realurl-decodeSpeakingURL' => [
+            'target' => \Tx\Realurl\Middleware\DecodeSpeakingURL::class,
+            'after' => [
+                'typo3/cms-frontend/page-resolver',
```

Comment:
this way, we have an issue because $TSFE->determineId() is called (within page-resolver) before the hook....

Reply:
Now the realurl middleware executes between the SiteResolver and the PageResolver.